Simplifying concurrency model with kotlin coroutines

brianmcgee opened the pull request:
# Description

Simplifies the concurrency model within the ingestors, replacing RxJava with Kotlin co-routines.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Manaul testing for now.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

brianmcgee committed:
feat: replace rxjava with coroutines in postgres ingestor

* removed the import queue
* much simpler concurrency model based on kotlin co-routines and channels

brianmcgee committed:
wip: simplifying kafka with co-routines

brianmcgee committed:
fix: respect processing level in postgres connector

brianmcgee committed:
fix: correctly handle staticcall when tracing

brianmcgee committed:
fix: setting isCanonical for block header correctly

brianmcgee committed:
fix: respect earliest block number config in postgres connector

brianmcgee committed:
chore: lint

brianmcgee committed:
doc: update usage.md

aldoborrero reviewed (commented):
@brianmcgee have you removed rxjava dependencies from `gradle/dependencies` and from being imported into `ingestion/base`?

brianmcgee commented:
> @brianmcgee have you removed rxjava dependencies from `gradle/dependencies` and from being imported into `ingestion/base`?

Not yet, I will clean up the dependencies before marking the PR as no longer a draft.

Commit:
feat: removed kafka support

Commit:
wip: added initial skeleton for testing PostgresBlockWriter

Commit:
feat: added minor corrections to PostgresBlockWriter and also the possibility to configure the poll time

Commit:
chore: removed unused method in BlockReader.kt

Commit:
wip: added improvements to PostgresBlockWriterSpec.kt

Commit:
chore: formatted source with ktlint

Commit:
wip: updated PostgresBlockWriterSpec

brianmcgee committed:
feat: block number based import for postgres

brianmcgee committed:
fix: touched accounts export for postgres connector

brianmcgee committed:
fix: remove start block override

Allowing a start block override does not make sense given that some state may be
dependent on state from previous blocks and results in foreign key constraints
failing on insert.

brianmcgee committed:
feat: improve postgres plugin logging

brianmcgee committed:
feat: initial implementation of simplified kafka connector

brianmcgee committed:
Revert "fix: remove start block override"

This reverts commit 4f88df48

brianmcgee committed:
feat: refine how processing level is specified

brianmcgee committed:
feat: implement earliest block for kafka writer

brianmcgee committed:
chore: remove kafka ingestor and lint

Moved kafka ingestor to a separate branch as it requires more work to complete well.

brianmcgee committed:
wip: add embedded postgres for testing
